A very simple test case is as below:
```
func TestContainerAndPodFromLabels(t *testing.T) {
        inspect := &dockertypes.ContainerJSON{}

        if inspect == nil {
                t.Errorf("inspect is nil")
        } else if inspect.Config == nil {
                t.Errorf("inspect.Config is nil")
        } else {
                t.Errorf("inspect.Config is not nil")
        }

        _, _, err := containerAndPodFromLabels(inspect)
        if err != nil {
                t.Errorf("unable to access pod data from container: %v", err)
        }
}
```

Then panic occurs!
```
--- FAIL: TestContainerAndPodFromLabels (0.00s)
	docker_manager_test.go:2224: inspect.Config is nil
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0xf0 pc=0x4a38df]

goroutine 158 [running]:
panic(0x186c860, 0xc820010110)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
testing.tRunner.func1(0xc82070bdd0)
	/usr/local/go/src/testing/testing.go:467 +0x192
panic(0x186c860, 0xc820010110)
	/usr/local/go/src/runtime/panic.go:426 +0x4e9
k8s.io/kubernetes/pkg/kubelet/dockertools.containerAndPodFromLabels(0xc82003df30, 0x0, 0x0, 0x0, 0x0)
	/home/paas/zxp/code/k8s/fork/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/dockertools/docker_manager.go:1442 +0xcf
k8s.io/kubernetes/pkg/kubelet/dockertools.TestContainerAndPodFromLabels(0xc82070bdd0)
	/home/paas/zxp/code/k8s/fork/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/dockertools/docker_manager_test.go:2229 +0xd2
testing.tRunner(0xc82070bdd0, 0x2707740)
	/usr/local/go/src/testing/testing.go:473 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:582 +0x892
FAIL	k8s.io/kubernetes/pkg/kubelet/dockertools	0.170s
!!! Error in hack/make-rules/test.sh:192
  'return ${rc}' exited with status 1
Call stack:
  1: hack/make-rules/test.sh:192 main(...)
Exiting with status 1
make: *** [test] Error 1
```

/cc @jfrazelle @vishh

--- a/pkg/kubelet/dockertools/docker_manager.go
+++ b/pkg/kubelet/dockertools/docker_manager.go
@@ -1434,7 +1434,7 @@ var errNoPodOnContainer = fmt.Errorf("no pod information labels on Docker contai
 
 // containerAndPodFromLabels tries to load the appropriate container info off of a Docker container's labels
 func containerAndPodFromLabels(inspect *dockertypes.ContainerJSON) (pod *api.Pod, container *api.Container, err error) {
-	if inspect == nil && inspect.Config == nil && inspect.Config.Labels == nil {
+	if inspect == nil || inspect.Config == nil || inspect.Config.Labels == nil {
 		return nil, nil, errNoPodOnContainer
 	}
 	labels := inspect.Config.Labels

--- a/pkg/kubelet/dockertools/docker_manager_test.go
+++ b/pkg/kubelet/dockertools/docker_manager_test.go
@@ -2305,3 +2305,35 @@ func TestSyncPodGetsPodIPFromNetworkPlugin(t *testing.T) {
 		"create", "start", "inspect_container",
 	})
 }
+
+// only test conditions "if inspect == nil || inspect.Config == nil || inspect.Config.Labels == nil" now
+func TestContainerAndPodFromLabels(t *testing.T) {
+	tests := []struct {
+		inspect       *dockertypes.ContainerJSON
+		expectedError error
+	}{
+		{
+			inspect:       nil,
+			expectedError: errNoPodOnContainer,
+		},
+		{
+			inspect:       &dockertypes.ContainerJSON{},
+			expectedError: errNoPodOnContainer,
+		},
+		{
+			inspect: &dockertypes.ContainerJSON{
+				Config: &dockercontainer.Config{
+					Hostname: "foo",
+				},
+			},
+			expectedError: errNoPodOnContainer,
+		},
+	}
+
+	for k, v := range tests {
+		pod, container, err := containerAndPodFromLabels(v.inspect)
+		if pod != nil || container != nil || err != v.expectedError {
+			t.Errorf("case[%q]: expected: nil, nil, %v, got: %v, %v, %v", k, v.expectedError, pod, container, err)
+		}
+	}
+}

